### PR TITLE
Change horizontal and vertical realignment to only occur when line is out of bounds.

### PR DESCRIPTION
**Pull request by mmcote**
Associated Issue: #5138 

Here's the Pull Request Doc
https://devtools-html.github.io/debugger.html/CONTRIBUTING.html#pull-requests

### Summary of Changes

-Check if the line is currently visible.
-Only realign the line to be centered if the line is out of bounds.

### Test Plan
N/A

Here's the Debugger's Testing doc
https://docs.google.com/document/d/1oBMRxV8A2ag2t22YsQOxTdEv0mXKzIg0tggJjRkU1S0/edit#.
Feel free to improve it!

Before: 
- As can be seen before, on every step in the debugger the editor was being realigned horizontally and vertically.
beforestepdebug

After:
- After the changes the editor will only be realigned when the line goes out of the bounds of the editor.
afterstepper

Before:
- Negative on the margin on the right side of the scroll area was increasing the clientWidth of the scrollArea.
screen shot 2018-01-23 at 6 49 05 pm

After:
- Removed extra margin on the right side of the scroll area.
screen shot 2018-01-23 at 6 49 21 pm

Also added some necessary functions to be mocked in Editor.spec.js, that are used in the isVisible function.

### CODE REVIEW

File: src/utils/editor/index.js [hunk 1]
```diff
@@ -87,11 +87,37 @@ export function scrollToColumn(codeMirror: any, line: number, column: number) {
...
+    scrollArea.left,
+    scrollArea.left + scrollArea.clientWidth - charWidth
+  );
+
+  const fontHeight = codeMirror.defaultTextHeight();
+  const inYView = withinBounds(
+    top,
+    scrollArea.top,
+    scrollArea.top + scrollArea.clientHeight - fontHeight
+  );
```
Comment (jasonLaster) [inline src/utils/editor/index.js]: how does this work?

File: src/components/Editor/Editor.css [hunk 2]
```diff
@@ -126,6 +126,10 @@ html[dir="rtl"] .editor-mount {
   height: 100%;
 }
 
+.CodeMirror-scroll {
+  margin-right: 0px;
+}
```
Comment (jasonLaster) [inline src/components/Editor/Editor.css]: why is this needed?

Comment (mmcote) [reply src/components/Editor/Editor.css]: When I was using the codeMirror, I noticed that the clientWidth that was being returned from the getScrollInfo() was not matching what was seen in the inspector. It was off by 30px. I found that by default it had a margin-right: -30px. Although once I removed this extra negative margin, it properly represented the scroll area, the bounds of the scroll view.

File: src/utils/editor/index.js (hunk 1 above)
Comment (mmcote) [reply src/utils/editor/index.js]: I did this just because if your scroll view area was not exactly a multiple of the font height then the last line may be cut off when you go to debug the last line, as it will still stay in view. So instead I said that if the top is below the top of the last lines fontHeight just realign and then you can see the whole line.

File: src/components/Editor/Editor.css (hunk 2 above)
Comment (lukaszsobek) [reply src/components/Editor/Editor.css]: @jasonLaster @mmcote you might find this discussion interesting: https://github.com/codemirror/CodeMirror/issues/2984
not the same margin, but maybe the same reason.

Comment (mmcote) [reply src/components/Editor/Editor.css]: Thank you for bringing this to my attention :) I will make some corresponding changes when I get a chance later on in the day.
